src/Git.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Composer\Satis;
+
+use Exception;
+
+class Git {
+    private $location = null;
+
+    public function __construct() {
+
+    }
+
+    public function setDir(string $location): Git {
+        $this->location = $location;
+
+        return $this;
+    }
+
+    public function getCollaborators() {
+        $resp = $this->runGitCommand("log --format='{\"name\": \"%aN\", \"email\": \"%aE\"}'  | sort |  uniq -c | sort -nr");
+
+        $lines = explode("\n", $resp);
+
+        $colabs = [];
+
+        foreach ($lines as $line) {
+            if ($line === "") {
+                continue;
+            }
+
+            $colabs[] = json_decode("{" . explode("{", $line)[1], true);
+        }
+
+        // Remove all array items with a duplicate email
+        $colabs = array_intersect_key(
+            $colabs,
+            array_unique(array_column($colabs, 'email'))
+        );
+
+        return $colabs;
+    }
+
+    public function runGitCommand(string $command) {
+        $cmds = [];
+
+        $cmds[] = "cd $this->location";
+        $cmds[] = "git --git-dir . --work-tree . $command";
```

Comment:
If you specifiy the `git-dir`, why first cd into that directory?

Reply:
Only the actual `.git` folders are available within the composer cache, not the actual project folders, so adding both `git-dir` and `work-tree` to the `.git` folder would fix this issue, if I were to remove these, it simply would throw out a repository couldn't be found.